mypy/subtypes.py

```diff
@@ -186,7 +186,10 @@ def visit_tuple_type(self, left: TupleType) -> bool:
                   is_named_instance(right, 'typing.Container') or
                   is_named_instance(right, 'typing.Sequence') or
                   is_named_instance(right, 'typing.Reversible')):
-                iter_type = right.args[0]
+                if right.args:
```

Comment:
This seems to just paper over the real bug which might still cause trouble elsewhere. `Sequence` without arguments should always be represented internally as `Sequence[Any]`, and the bug is with letting the subtype check see a broken `Sequence` type.